### ISSUE TITLE
[bug]: Trying to FUND ACCOUNT and receive the error "UNABLE TO CREATE THE ACCOUNT no_route"

### ISSUE THREAD

**Issue by machadommm** [lightningnetwork/lnd#7897]
Dear,

I'm trying to create a POOL account but I'm getting the error below.
I opened a ticket in the POOL repository and with their help it was detected that the node does not reach the entire LNnetwork.

The opened ticket was this one: https://github.com/lightninglabs/pool/issues/464

Even having an open channel from A to node B and this connected to node C, node A does not reach node C.

Below is the output from the problem node:

lndcli getnetworkinfo
{
    "graph_diameter": 15,
    "avg_out_degree": 3.6692667706708266,
    "max_out_degree": 116,
    "num_nodes": 12179,
    "num_channels": 22344,
    "total_network_capacity": "94757201717",
    "avg_channel_size": 4240834.305272109,
    "min_channel_size": "1050",
    "max_channel_size": "1000000000",
    "median_channel_size_sat": "1000000",
    "num_zombie_chans": "183126"
}

And on this another node in a location with no problem:
lndcli getnetworkinfo
{
    "graph_diameter": 13,
    "avg_out_degree": 4.831333579998943,
    "max_out_degree": 83,
    "num_nodes": 18919,
    "num_channels": 45702,
    "total_network_capacity": "158219388275",
    "avg_channel_size": 3461979.5255131065,
    "min_channel_size": "1050",
    "max_channel_size": "1000000000",
    "median_channel_size_sat": "1000000",
    "num_zombie_chans": "143523"
}

Those nodes above are under my administration and share the same configuration.

**Comment by machadommm** [comment]
Hi everyone,
Any response about this problem?

Cheers,

**Comment by Redinpais** [comment]
I'm struggling with the same problem.
Here's a printout of my result compared to what Amboss tells me.
![Captura de ecrã 2023-09-04 173224](https://github.com/lightningnetwork/lnd/assets/17316352/46073998-ca7e-422b-bccb-5a9b91d45a7d)

**Comment by diegokolling** [comment]
Well, yeah, here I have the same problem:

{
    "graph_diameter": 14,
    "avg_out_degree": 4.9888008740781205,
    "max_out_degree": 108,
    "num_nodes": 10983,
    "num_channels": 27396,
    "total_network_capacity": "135104638226",
    "avg_channel_size": 4931546.146371733,
    "min_channel_size": "1050",
    "max_channel_size": "1000000000",
    "median_channel_size_sat": "1000000",
    "num_zombie_chans": "220891"
}

A regular, up to date Umbrel user in hybrid mode using TunnelSats VPN.

**Comment by Juka-Bala** [comment]
I have the same problem, too:
![image](https://github.com/lightningnetwork/lnd/assets/122131936/e4f1dc5f-c751-4905-b086-fa660c158356)

**Comment by guggero** [comment]
Folks, you can't just take Amboss' numbers and compare that to your local graph. Amboss collects _all_ information of every channel it ever sees and doesn't do any cleanup. `lnd` will remove channels it doesn't see an update for 2 weeks (this is what is called "zombie pruning") so the graph can be small, efficient and fast when doing pathfinding.
Also, these numbers rely on your setting of `routing.strictgraphpruning`, so when pasting screenshots, please also mention the state of that configuration flag.

Also, because every node's graph is assembled from the gossip messages it receives, it can look different than other nodes (like there is not one single mempool but each node's mempool).

That being said, it does look like some of those graph numbers look a bit off. So there is something to look into here.

**Comment by machadommm** [comment]
Hi Guggero,
Look at the checks of these nodes below:

1 - Node T4b4j4r4ORG with routing.strictgraphpruning=1, Active Channels: 23 and hybrid network:

machado@t4b4node:~$ lndcli getnetworkinfo
{
     "graph_diameter": 13,
     "avg_out_degree": 4.764941515782727,
     "max_out_degree": 97,
     **"num_nodes": 18723,**
     "num_channels": 44607,
     "total_network_capacity": "162574217005",
     "avg_channel_size": 3644589.795435694,
     "min_channel_size": "1050",
     "max_channel_size": "1000000000",
     "median_channel_size_sat": "1000000",
     "num_zombie_chans": "146247"
}

2 - Node T4m010ORG with routing.strictgraphpruning=0, Active Channels: 31 and hybrid network:
machado@jginyuen0d3:~$ lndcli getnetworkinfo
{
     "graph_diameter": 15,
     "avg_out_degree": 3.732225148123766,
     "max_out_degree": 122,
     **"num_nodes": 12152,**
     "num_channels": 22677,
     "total_network_capacity": "99320952491",
     "avg_channel_size": 4379810.049433347,
     "min_channel_size": "1050",
     "max_channel_size": "1000000000",
     "median_channel_size_sat": "1000000",
     "num_zombie_chans": "183820"
}

3 - Node 0203ddab15712271ce19 with routing.strictgraphpruning=0,  Active Channels = 0 and onion only network (this node is a test one, recently installed, in the same physical network as the node T4m010ORG that has the problem):
machado@jginyueadm:~$ lndcli getnetworkinfo
{
     "graph_diameter": 13,
     "avg_out_degree": 4.735827895595432,
     "max_out_degree": 455,
     **"num_nodes": 19616,**
     "num_channels": 46449,
     "total_network_capacity": "151411504556",
     "avg_channel_size": 3259736.583263364,
     "min_channel_size": "1050",
     "max_channel_size": "1000000000",
     "median_channel_size_sat": "1000000",
     "num_zombie_chans": "129"
}

Notice that the node with the problem is number 2.

Is there any specific command in LND that we can extract the settings applied while the node is running, that is, like a "show running config" command?

Thanks,

**Comment by Roasbeef** [comment]
@machadommm how many peers/channels does your node have? If you disable `strictgraphpruning` on that node, and wait a few hours to sync up, does your graph start to populate?

**Comment by machadommm** [comment]
Hi @Roasbeef I have 29 channels (https://amboss.space/node/02d6f4112fdb133ed2c26a5617d6c996526d3eed45dc0711da4dd71e56d4641d8c) and I did the test disabling  strictgraphpruning and retrived same result. Not happen.

**Comment by guggero** [comment]
@machadommm we think we have a way to force a graph re-sync, see https://github.com/lightningnetwork/lnd/pull/7988.
In order to test this without running an experimental version, I also added the command as a standalone feature of `chantools`.

Can you please try the following?
1. Install the `master` branch of `chantools` by following this: https://github.com/lightninglabs/chantools#compile-from-source (the new functionality is not yet in any released version, so you can't use pre-built binaries).
2. Stop `lnd`
3. Run `chantools dropgraphzombies --channeldb <path_to_your_channel_db>`
4. Start `lnd` again and wait long enough to see `synced_to_graph` turn `true` in `lncli getinfo`.

After that it would be very interesting to see the output of your `lncli getnetworkinfo` call.

**Comment by machadommm** [comment]
Hi @guggero I followed the indicated procedures, but it did not change the situation regarding the node's network reach.

machado@jginyuen0d3:~$ go/bin/chantools dropgraphzombies --channeldb /home/machado/umbrel/app-data/lightning/data/lnd/data/graph/mainnet/channel.db
2023-09-20 18:17:34.029 [INF] CHAN: chantools version v0.11.3 commit 
2023-09-20 18:17:34.337 [INF] CHDB: Checking for schema update: latest_version=29, db_version=29
2023-09-20 18:17:34.337 [INF] CHDB: Checking for optional update: prune_revocation_log=false, db_version=empty
2023-09-20 18:17:34.337 [INF] CHAN: Dropping zombie channel bucket

machado@jginyuen0d3:~$ lndcliente getinfo|grep synced_to_graph
    "synced_to_graph": true,

machado@jginyuen0d3:~$ lndcliente getnetworkinfo
{
    "graph_diameter": 16,
    "avg_out_degree": 3.909990574929312,
    "max_out_degree": 217,
    "num_nodes": 12732,
    "num_channels": 24891,
    "total_network_capacity": "115252889364",
    "avg_channel_size": 4630303.698686272,
    "min_channel_size": "1050",
    "max_channel_size": "1000000000",
    "median_channel_size_sat": "1000000",
    "num_zombie_chans": "11976"
}

For information, the log showed these errors below during the synchronization process:

2023-09-20 21:30:24.240 [ERR] CRTR: process network updates got: unable to fetch utxo for chan_id=677100151192748033, chan_point=a579603836f750a44f526c41d570b646064de40dfb4bf457f7f74f87b07d7500:1: target output has been spent
2023-09-20 21:30:24.342 [ERR] CRTR: process network updates got: unable to fetch utxo for chan_id=677119942467059713, chan_point=1d9d9c91c228d367da29656fe1e89ffbc2a6a4716333a4dcff502a3a7d6597b2:1: target output has been spent
2023-09-20 21:30:24.435 [ERR] CRTR: process network updates got: unable to fetch utxo for chan_id=677123240990605312, chan_point=ebe283e139e9781fef5181d66679d00dd9a567f2cf4555640f5a2d4739a207ad:0: target output has been spent
2023-09-20 21:30:24.540 [ERR] CRTR: process network updates got: unable to fetch utxo for chan_id=677126539508187136, chan_point=2a3893634cf36df381b115dd7c3bcd249d35dd5a776d9479e134841b9355b953:0: target output has been spent
2023-09-20 21:30:24.647 [ERR] CRTR: process network updates got: unable to fetch utxo for chan_id=677126539532959744, chan_point=e9c527f6fcd0871fbe4f8be179d51a1d955371015f7a6c83b52e26da96f2eed7:0: target output has been spent
2023-09-20 21:30:24.744 [ERR] CRTR: process network updates got: unable to fetch utxo for chan_id=677137534566203392, chan_point=cd05ac7c12530edbd3d71e39bb976c403b225a076872b949e2860eec8ac870f7:0: target output has been spent
2023-09-20 21:30:24.838 [ERR] CRTR: process network updates got: unable to fetch utxo for chan_id=677147430214041601, chan_point=e90a726f69b66340a1005bf51e352f9a997be580d1eacb52d31abb3e4925ca7a:1: target output has been spent
2023-09-20 21:30:24.953 [ERR] CRTR: process network updates got: unable to fetch utxo for chan_id=677154027283808256, chan_point=5b046c1254f9a674fe181b31640612478345ba1e6422f901c663c4d944818890:0: target output has been spent
2023-09-20 21:30:25.045 [ERR] CRTR: process network updates got: unable to fetch utxo for chan_id=675371718835044353, chan_point=64f2256cd37ddb8af0e77df8e503f62c89d3ebf6cedef4550adb794c765df5bf:1: target output has been spent
2023-09-20 21:30:25.215 [ERR] CRTR: process network updates got: unable to fetch utxo for chan_id=677159524828905473, chan_point=0932cfce65736a469e107eba406e197d2fb62f94db72d6339b5d8a637ad534bd:1: target output has been spent

**Comment by machadommm** [comment]
Hi @guggero With the failure of the above attempt, I went back to reviewing the entire connection with my Internet providers.
As I have two links to the Internet, I forced my node to exit only through one of them and with that the network started to reappear for the node.
As I use the node in hybrid mode, there may be some trouble  with multiple Internet connections to the onion network.

The result I got now was this:

machado@jginyuen0d3:~$ lndcliente getnetworkinfo
{
    "graph_diameter": 13,
    "avg_out_degree": 4.624645971011273,
    "max_out_degree": 135,
    **"num_nodes": 18007,**
    "num_channels": 41638,
    "total_network_capacity": "143787428401",
    "avg_channel_size": 3453274.134228349,
    "min_channel_size": "1050",
    "max_channel_size": "514987164",
    "median_channel_size_sat": "1000000",
    "num_zombie_chans": "11978"
}

So I believe this issue has been resolved for me. I will close the ticket and if you want more details, I am available to collaborate.

Thank you very much for your help and attention to the matter.
